src/org/spdx/licenselistpublisher/LicenseRDFAGenerator.java

```diff
@@ -173,15 +174,34 @@ public static void main(String[] args) {
 		}
 		String[] ignoredWarnings = new String[0];
 		if (args.length > 5) {
-			CSVReader reader = null;
+			Reader reader = null;
+			Gson gson = new Gson();
+			List<List<String>> jsonWarnings = new ArrayList<>();
 			try {
+				// read JSON array of array of paired strings
 				File warningsFile = new File(args[5]);
 				if (warningsFile.exists()) {
-					reader = new CSVReader(new FileReader(warningsFile));
+					reader = new FileReader(warningsFile);
 				} else {
-					reader = new CSVReader(new StringReader(args[5]));
+					reader = new StringReader(args[5]);
```

Comment:
I should note specifically that, although I retained this second branch which switches the argument to a "StringReader", I did not actually test this and I'm not clear on the use case.

I _assume_ that it's meant to enable providing a command-line argument with the actual expected-warnings text, rather than the argument being a path to the file containing that text.

If that's correct, then I am guessing this may still work as long as that command-line argument is switched to be JSON instead (and would need to be an array of arrays of strings, e.g. `[ ["GPL-2.0", "GPL-2.0+"], ... ]`  But again, I haven't actually tested this so feedback is very welcome.  :)